public/app/features/trails/DataTrail.tsx

```diff
@@ -137,8 +137,8 @@ export class DataTrail extends SceneObjectBase<DataTrailState> {
       );
     }
 
-    // Save the current trail as a recent if the browser closes or reloads
-    const saveRecentTrail = () => getTrailStore().setRecentTrail(this);
+    // Save the current trail (if it has a metric) as a recent if the browser closes or reloads
+    const saveRecentTrail = () => this.state.metric && getTrailStore().setRecentTrail(this);
```

Comment:
It's quite a nice solution to prevent saving here in one place. But without a metric we also prevent saving a search with bunch of filters and time range applied. I don't think we want that. Please verify this behavior is the desired one. If not you can check the history step count and if it is just one that means it is a fresh selection and just don't save.

Reply:
ah great catch, thank you! will address when pairing with brendan tomorrow